suite/tests/CMakeLists.txt

```diff
@@ -2414,12 +2433,21 @@ endif ()
       # i#3348: Ensure that DR without local symbols hidden is linkable and
       # avoids conflicts (though CMake_symbol_check does most of the conflict
       # checking).
-      tobuild_api(api.static_symbols api/static_symbols.c "" "" OFF ON)
-      get_target_property(lib_list api.static_symbols LINK_LIBRARIES)
-      list(REMOVE_ITEM lib_list dynamorio_static)
-      set_property(TARGET api.static_symbols PROPERTY LINK_LIBRARIES ${lib_list})
-      target_link_libraries(api.static_symbols dynamorio_static_nohide)
+      tobuild_static_nohide_api(api.static_symbols api/static_symbols.c "" "" OFF)
       append_link_flags(api.static_symbols "-Wl,--warn-common -Wl,--fatal-warnings")
+
+      # To ensure that the dynamorio_so_start/_end linker variables are actually
+      # used in static binaries, we have two tests which run the same "map mixup"
+      # logic: one which exits successfully because we provide the linker
+      # variables and another which exits with an assertion because we don't.
+      # Both versions utilize the dynamorio_static_nohide as a base because
+      # the variables in dynamorio_static have already been resolved as a part of
+      # the symbol hiding step.
+      set(api.static_maps_mixup_yesvars_expectbase "static_maps_mixup_yesvars")
+      tobuild_static_nohide_api(api.static_maps_mixup_yesvars api/static_maps_mixup.c "" "" OFF)
+      append_link_flags(api.static_maps_mixup_yesvars "-Wl,--defsym,dynamorio_so_start=__executable_start -Wl,--defsym,dynamorio_so_end=end")
```

Comment:
style: newline before " will get under 80 chars?  (No enforcement in cmake files, just best manual effort)

Reply:
Done.